src/rgw/rgw_putobj_processor.cc

```diff
@@ -105,6 +118,7 @@ int RadosWriter::write_exclusive(const bufferlist& data)
   const uint64_t cost = data.length();
 
   librados::ObjectWriteOperation op;
+  add_write_hint(op);
```

Comment:
moving this call down one line, below `op.create(true); // exclusive create`, appears to resolve the failures

Reply:
done. Please test it again. Thanks